### PR TITLE
Fix Query Param Passing

### PR DESCRIPTION
**Pull request by svanherk**
- We _do_ want the load function to run each time the entity changes
- Using a helper function until query parameter parsing is fixed for performSirenAction

### CODE REVIEW

File: components/d2l-hm-filter/d2l-hm-filter.js [hunk 1]
```diff
@@ -405,6 +405,26 @@ class D2LHypermediaFilter extends mixinBehaviors([D2L.PolymerBehaviors.Siren.Ent
 			)
 		);
 	}
+
+	/* Helper needed until we have fixed the functionality in SirenActionBehavior
+	* Specifically, the getSirenFields function is broken: https://github.com/Brightspace/polymer-siren-behaviors/blob/master/store/siren-action-behavior.js#L14
+	* It does not grab the query parameters correctly, and duplicate parameters and fields should not be included
+	*/
+	_performSirenActionWithQueryParams(action) {
+		const url = new URL(action.href, window.location.origin);
```
Comment (svanherk) [inline components/d2l-hm-filter/d2l-hm-filter.js]: I need to confirm that BSI will handle this for us in IE11, or add that

File: components/d2l-hm-filter/d2l-hm-filter.js [hunk 2]
```diff
@@ -405,6 +405,26 @@ class D2LHypermediaFilter extends mixinBehaviors([D2L.PolymerBehaviors.Siren.Ent
...
+
+		if (!action.fields) {
+			action.fields = [];
+		}
+
+		url.searchParams.forEach(function(value, key) {
+			if (!action.fields.filter(x => x.name === key)[0]) {
+				action.fields.push({ name: key, value: value, type: 'hidden' });
```
Comment (AlexBedley) [inline components/d2l-hm-filter/d2l-hm-filter.js]: is `hidden` necessary?

File: components/d2l-hm-filter/d2l-hm-filter.js [hunk 3]
```diff
@@ -405,6 +405,26 @@ class D2LHypermediaFilter extends mixinBehaviors([D2L.PolymerBehaviors.Siren.Ent
...
+	*/
+	_performSirenActionWithQueryParams(action) {
+		const url = new URL(action.href, window.location.origin);
+
+		if (!action.fields) {
+			action.fields = [];
+		}
+
+		url.searchParams.forEach(function(value, key) {
+			if (!action.fields.filter(x => x.name === key)[0]) {
```
Comment (AlexBedley) [inline components/d2l-hm-filter/d2l-hm-filter.js]: You can use Carol's ie11 find hack `_findInArray`

File: components/d2l-hm-filter/d2l-hm-filter.js (hunk 2 above)
Comment (AlexBedley) [reply components/d2l-hm-filter/d2l-hm-filter.js]: `performSirenAction` takes a second parameter called `fields` that could also work. But I think you're more familiar with the internals of `performSirenAction` than I am so I leave it up to you

File: components/d2l-hm-filter/d2l-hm-filter.js [hunk 4]
```diff
@@ -405,6 +405,26 @@ class D2LHypermediaFilter extends mixinBehaviors([D2L.PolymerBehaviors.Siren.Ent
 			)
 		);
 	}
+
+	/* Helper needed until we have fixed the functionality in SirenActionBehavior
+	* Specifically, the getSirenFields function is broken: https://github.com/Brightspace/polymer-siren-behaviors/blob/master/store/siren-action-behavior.js#L14
+	* It does not grab the query parameters correctly, and duplicate parameters and fields should not be included
+	*/
+	_performSirenActionWithQueryParams(action) {
```
Comment (AlexBedley) [inline components/d2l-hm-filter/d2l-hm-filter.js]: I definitely want to see unit tests for this guy

File: components/d2l-hm-filter/d2l-hm-filter.js [hunk 5]
```diff
@@ -106,7 +106,7 @@ class D2LHypermediaFilter extends mixinBehaviors([D2L.PolymerBehaviors.Siren.Ent
 	}
 
 	async _loadFilters(entity) {
-		if (!entity || (this._filters && this._filters.length)) {
+		if (!entity) {
```
Comment (AlexBedley) [inline components/d2l-hm-filter/d2l-hm-filter.js]: What's the reason behind this change?

File: components/d2l-hm-filter/d2l-hm-filter.js (hunk 4 above)
Comment (svanherk) [reply components/d2l-hm-filter/d2l-hm-filter.js]: We're also going to be adding this to the eval hub list and Jamie was working on unit tests.  I can sync up with him and we can add it in both places

File: components/d2l-hm-filter/d2l-hm-filter.js (hunk 5 above)
Comment (svanherk) [reply components/d2l-hm-filter/d2l-hm-filter.js]: Originally, we were saying "if this isn't the initial load and we've already loaded filters, skip this and do nothing" as a perf gain.  Which completely breaks the whole idea of reloading the filters and their info and self links when the table changes or the href of eval hub.  We want to reload everything if our entity ever changes

File: components/d2l-hm-filter/d2l-hm-filter.js (hunk 2 above)
Comment (svanherk) [reply components/d2l-hm-filter/d2l-hm-filter.js]: I totally stole this from @jmurray87.  Jamie, did you find `hidden` necessary, and was there any reasoning between choosing this or using the second parameter?

Comment (jmurray87) [reply components/d2l-hm-filter/d2l-hm-filter.js]: I added it because of this line https://github.com/Brightspace/polymer-siren-behaviors/blob/bed996403a40dbb92675224df4fa2dc25e58ba5f/store/siren-action-behavior.js#L150 If we're going to add the query params as fields manually, I didn't want the action behaviour library to try to grab from url

Comment (AlexBedley) [reply components/d2l-hm-filter/d2l-hm-filter.js]: I know there is at least 1 usage of `performSirenAction` where we add fields (in the master teacher stuff). Just wanna make sure it works in tandem with that.

Comment (jmurray87) [reply components/d2l-hm-filter/d2l-hm-filter.js]: We aren't modifying that call, that one is on the enrollments domain that won't have to worry about existing url param state

File: components/d2l-hm-filter/d2l-hm-filter.js (hunk 1 above)
Comment (svanherk) [reply components/d2l-hm-filter/d2l-hm-filter.js]: Confirmed that it does not, so removed all non-IE11-compatible things

**Comment by svanherk** [comment]
Updated to include no non-compatible IE11 code (other than the siren helpers themselves, grr) since this is a shared component and added tests.  Ready for re-review!
